single-line-diagram-force-layout/src/main/java/com/powsybl/sld/force/layout/ForceLayout.java

```diff
@@ -0,0 +1,256 @@
+package com.powsybl.sld.force.layout;
+
+import org.jgrapht.Graph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+
+public class ForceLayout<V, E extends Spring> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ForceLayout.class);
+
+    private final Random random = new Random();
+
+    private static final int DEFAULT_MAX_STEPS = 2000;
+    private static final double DEFAULT_MIN_ENERGY_THRESHOLD = 0.01;
+    private static final double DEFAULT_DELTA_TIME = 0.05;
+    private static final double DEFAULT_REPULSION = 400.0;
+    private static final double DEFAULT_DAMPING = 0.5;
+    private static final double DEFAULT_MAX_SPEED = Double.POSITIVE_INFINITY;
+
+    private int maxSteps;
+    private double minEnergyThreshold;
+    private double deltaTime;
+    private double repulsion;
+    private double damping;
+    private double maxSpeed;
+
+    private Graph<V, E> graph;
+    private Map<V, Point> points;
+
+    public ForceLayout(Graph<V, E> graph) {
+        this.maxSteps = DEFAULT_MAX_STEPS;
+        this.minEnergyThreshold = DEFAULT_MIN_ENERGY_THRESHOLD;
+        this.deltaTime = DEFAULT_DELTA_TIME;
+        this.repulsion = DEFAULT_REPULSION;
+        this.damping = DEFAULT_DAMPING;
+        this.maxSpeed = DEFAULT_MAX_SPEED;
+
+        this.graph = graph;
+        this.points = new HashMap<>();
+    }
+
+    public ForceLayout<V, E> setMaxSteps(int maxSteps) {
+        this.maxSteps = maxSteps;
+        return this;
+    }
+
+    public ForceLayout<V, E> setMinEnergyThreshold(double minEnergyThreshold) {
+        this.minEnergyThreshold = minEnergyThreshold;
+        return this;
+    }
+
+    public ForceLayout<V, E> setDeltaTime(double deltaTime) {
+        this.deltaTime = deltaTime;
+        return this;
+    }
+
+    public ForceLayout<V, E> setRepulsion(double repulsion) {
+        this.repulsion = repulsion;
+        return this;
+    }
+
+    public ForceLayout<V, E> setDamping(double damping) {
+        this.damping = damping;
+        return this;
+    }
+
+    public ForceLayout<V, E> setMaxSpeed(double maxSpeed) {
+        this.maxSpeed = maxSpeed;
+        return this;
+    }
+
+    public ForceLayout<V, E> setInitialisationSeed(long seed) {
+        random.setSeed(seed);
+        return this;
+    }
+
+    // TODO: implement other initialisations methods
+    public ForceLayout<V, E> initializePoints() {
+        for (V vertex : this.graph.vertexSet()) {
+            // TODO: not sure vertex is unique anymore if the user override the equals method
```

Comment:
As it is a set, you can be sure vertex is unique, as the set uses the equals method to contiain each element only once